### PR TITLE
Udp events and tests

### PR DESCRIPTION
**Pull request by antonbobkov**
generic hook into udp server and more tests

### CODE REVIEW

File: rebar_eunit_run [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #/bin/bash
 
-rebar3 eunit -mpop_verifier_test
\ No newline at end of file
+rebar3 eunit -mpotato_udp_test
```
Comment (peterluthunder) [inline rebar_eunit_run]: :+1:

File: src/potato_udp.erl [hunk 2]
```diff
@@ -16,62 +16,91 @@
 %% maps node_id to Pid of the process
 -type node_map() :: map().
 
--type state() :: {gen_udp:socket(), node_map()}.
+-type event_fn() :: fun( (atom(), any()) -> any() ).
 
+-record(udp_state,
```
Comment (peterluthunder) [inline src/potato_udp.erl]: :o

File: src/potato_udp.erl [hunk 3]
```diff
@@ -16,62 +16,91 @@
...
+	  socket :: gen_udp:socket(),
+	  node_map :: node_map(),
+	  event_fn :: event_fn()
+	}).
 
-start_link(Port) ->
-    gen_server:start_link({local, potato_udp}, potato_udp, Port, []).
 
--spec init(udp_port()) -> {ok, state()}.
+-type state() :: #udp_state{}.
```
Comment (peterluthunder) [inline src/potato_udp.erl]: :oo

Comment (antonbobkov) [reply src/potato_udp.erl]: was too lazy to change state() to #udp_state{}

Comment (antonbobkov) [reply src/potato_udp.erl]: such abstraction

File: rebar_eunit_run (hunk 1 above)
Comment (antonbobkov) [reply rebar_eunit_run]: too lazy to type it out each time

Comment (antonbobkov) [reply rebar_eunit_run]: also too lazy to wait for all other unit tests

File: test/potato_udp_test.erl [hunk 4]
```diff
@@ -3,30 +3,78 @@
...
+    
 
-potato_udp_test() ->
+start_stop_test() ->
     Port = 3142,
-    {ok, Pid} = gen_server:start(potato_udp, Port, []),
-    %%?debugFmt("started potato_udp gen_server ~p~n",[Pid]),
+    MyPid = self(),
+
+    ForwardFn = fun(Code, Data) -> MyPid ! {Code, Data} end,
```
Comment (antonbobkov) [inline test/potato_udp_test.erl]: I should really proper OTP event manage thing

mb i'll do it next

File: rebar_eunit_run (hunk 1 above)
Comment (peterluthunder) [reply rebar_eunit_run]: :+1:  is really just for the potato

**Comment by antonbobkov** [comment]
ignore this..

**Pull request merged**